contrib/templates/data-engineering/base/template/{{.project_name}}/scripts/test.py

```diff
@@ -0,0 +1,15 @@
+#!/usr/bin/env python3
+#
+# test.py runs the unit tests for this project using pytest.
+#
+import os
+import subprocess
+
+
+def main():
+    os.environ["DATABRICKS_SERVERLESS_COMPUTE_ID"] = "auto"
+    subprocess.run(["pytest"], check=True)
```

Comment:
If you only need to set this env var then you can use https://github.com/pytest-dev/pytest-env

The benefit of this is that it works out of the box with the in-IDE test runners/debuggers.

Reply:
I tested this with the following in my `pyproject.toml`:

```toml
[tool.pytest_env]
DATABRICKS_SERVERLESS_COMPUTE_ID = "auto"
```

And then I can run `uv run pytest`.

Reply:
`uv run test` is a shortcut for testing with serverless, but not everyone will want to run with serverless. Classic compute is still very common. So `uv run pytest` just lets you use your local configuration in `.databrickscfg`. Which might be serverless, but it might not be. When not using serverless, this tends to be highly personal and can't really fit in pyproject.toml.

Reply:
Maybe to add to the above: the intention here is that `pytest` simply follow's the configuration of the specific user and the VS Code plugin. `uv run test` is a shortcut that ignores your personal configuration and is serverless-only.